app/src/parser.c

```diff
@@ -27,13 +29,22 @@ void __assert_fail(const char * assertion, const char * file, unsigned int line,
 }
 #endif
 
+parser_tx_t parser_state;
+
 parser_error_t parser_parse(parser_context_t *ctx, const uint8_t *data, size_t dataLen) {
-    CHECK_PARSER_ERR(parser_init(ctx, data, dataLen))
-    return _read(ctx, &parser_tx_obj);
+    parser_state.state = NULL;
+    parser_state.len = 0;
+    // Just in case
+    //zb_deallocate();
+    CHECK_PARSER_ERR(_parser_init(ctx, data, dataLen, &parser_state.len))
+    if ( parser_state.len> 0 && zb_allocate(parser_state.len) == zb_no_error) {
+        zb_get(&parser_state.state);
```

Comment:
can we check the return value? Just in case something fails?

Reply:
done

Reply:
done